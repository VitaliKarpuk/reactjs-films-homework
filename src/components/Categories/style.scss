@import '../../utils.scss';

.movie-list__categories {
  @include flex;
  max-width: 100%;
  padding: 30px 30px;
  background-color: #e4e4e4;

  & > div {
    width: 100%;
    @include flex-between;
    border-bottom: 1px solid #acacac;

    & > div {
      width: 90px;
      @include flex-between;
    }
  }

  ul {
    @include flex-between;
    list-style: none;
    margin: 0;
    padding: 0;

    li {
      letter-spacing: 1px;
      line-height: 44px;
      font-weight: bold;
      color: #0aaee4;
      position: relative;

      a {
        letter-spacing: 1px;
        line-height: 44px;
        color: #0aaee4;
        font-weight: bold;
        text-decoration: none;
        padding: 10px;
      }

      &:hover {
        cursor: pointer;
      }
      .movie-list__categories-genre_list {
        display: flex;
        flex-direction: column;
        background-color: rgb(228, 228, 228);
        position: absolute;
        z-index: 1;
        padding: 10px;
        left: -9999px;
        border: 1px solid;
        border-top: none;
        transition: all 0.3s;
        -webkit-transition: all 0.3s;
        -moz-transition: all 0.3s;
        -ms-transition: all 0.3s;
        -o-transition: all 0.3s;

        li {
          border-bottom: 1px solid;
          width: 300px;
          font-size: 25px;

          &:hover {
            color: #004961;
          }
        }
      }

      &:hover .movie-list__categories-genre_list {
        left: 0;
      }
    }

    .categories_active-item {
      border-bottom: 4px solid #0aaee4;
    }
  }

  .movie-list__response_active,
  .movie-list__response {
    @include flex;
    flex-wrap: wrap;
    width: 35px;
    height: 35px;
    border-radius: 2px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    background-color: #3bb4e0;
    border-radius: 2px;
    position: relative;

    &::before {
      content: "";
      width: 100%;
      height: 5px;
      background-color: #e4e4e4;
    }

    &::after {
      content: "";
      width: 5px;
      height: 100%;
      background-color: #e4e4e4;
      position: absolute;
      left: 15px;
    }

    &:hover {
      cursor: pointer;
    }
  }

  .movie-list__response {
    background-color: #acacac;
  }

  .movie-list__no-response_active,
  .movie-list__no-response {
    @include flex;
    flex-wrap: wrap;
    width: 35px;
    height: 35px;
    background-color: #acacac;
    border-radius: 2px;

    &::before {
      content: "";
      width: 100%;
      height: 5px;
      background-color: #e4e4e4;
    }

    &:hover {
      cursor: pointer;
    }
  }

  .movie-list__no-response {
    background-color: #3bb4e0;
  }
}
