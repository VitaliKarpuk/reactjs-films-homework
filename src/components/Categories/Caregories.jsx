import React, { useState, useEffect } from "react";
import PropTypes from 'prop-types';
import { NavLink, useLocation } from "react-router-dom";
import { useSelector } from "react-redux";
import { categories } from "../../modules/constants/constants";
import { selectorGenre } from "../../modules/selectors/index";

import "./style.scss";

const Categories = ({ handleMovieListView, movieListView }) => {
  const [activeLink, setActiveLink] = useState("Trending");
  const arrResult = useSelector(state => selectorGenre(state));
  const location = useLocation();

  useEffect(() => {
    if (location.pathname) {
      setActiveLink(location.pathname);
    }
  }, [location]);


  return (
    <div className="movie-list__categories">
      <div>
        <ul>
          <li
            className={activeLink?.length === 1 ? "categories_active-item" : ""}
          >
            <NavLink exact to="/">
              Trending
            </NavLink>
          </li>
          <li
            className={
              activeLink?.includes("rated") ? "categories_active-item" : ""
            }
          >
            <NavLink to={`${categories.topRated}`}>Top Rated</NavLink>
          </li>
          <li
            className={
              activeLink?.includes("coming") ? "categories_active-item" : ""
            }
          >
            <NavLink to={`${categories.comingSoon}`}>Coming Soon</NavLink>
          </li>
          <li
            className={
              activeLink?.includes("genre")
                ? "movie-list__categories-genre categories_active-item"
                : "movie-list__categories-genre"
            }
          >
            <span>Genre &#8744;</span>
            <ul className="movie-list__categories-genre_list">
              {arrResult.map(item => {
                return (
                  <li key={item.id}>
                    <NavLink to={`${categories.genre}/${item.name}`}>
                      {item.name}
                    </NavLink>
                  </li>
                );
              })}
            </ul>
          </li>
        </ul>
        <div>
          <div
            className={
              movieListView ? "movie-list__response_active" : "movie-list__response"
            }
            onClick={handleMovieListView}
          >
            <span></span>
            <span></span>
            <span></span>
            <span></span>
          </div>
          <div
            className={
              movieListView
                ? "movie-list__no-response_active"
                : "movie-list__no-response"
            }
            onClick={handleMovieListView}
          >
            <span></span>
            <span></span>
          </div>
        </div>
      </div>
    </div>
  );
};

Categories.propTypes = {
  handleMovieListView: PropTypes.func,
  movieListView: PropTypes.bool
}

Categories.defaultProps = {
  handleMovieListView: () => {},
  movieListView: true
};

export default Categories;
