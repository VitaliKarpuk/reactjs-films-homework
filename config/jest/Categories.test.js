/* eslint-disable no-undef */
import React from 'react';
import Categories from '../../src/components/Categories/Caregories';

import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import cofigureMockStore from 'redux-mock-store';
import { MemoryRouter } from "react-router-dom";


const mockStore = cofigureMockStore();

describe('Test ModalInfo component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      films: [{ id: 1, genre_ids: [1] }],
      genre: [{ id: 1 }],
      dispatch: jest.fn()
    });

    component = renderer.create(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/top_rated']}>
          <Categories />
        </MemoryRouter>
      </Provider>
    );
  });
  it("renders Search component", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
})