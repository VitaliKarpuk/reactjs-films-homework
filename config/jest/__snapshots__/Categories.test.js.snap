// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ModalInfo component renders Search component 1`] = `
<div
  className="movie-list__categories"
>
  <div>
    <ul>
      <li>
        <a
          aria-current={null}
          href="/"
          onClick={[Function]}
        >
          Trending
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          className="categories_active-item"
          href="/top_rated"
          onClick={[Function]}
          style={Object {}}
        >
          Top Rated
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/upcoming"
          onClick={[Function]}
        >
          Coming Soon
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/genre"
          onClick={[Function]}
        >
          Genre âˆ¨
        </a>
        <ul
          className="movie-list__categories-genre_list"
        >
          <li>
            <a
              aria-current={null}
              href="/genre/undefined"
              onClick={[Function]}
            />
          </li>
        </ul>
      </li>
    </ul>
    <div>
      <div
        className="movie-list__response_active"
        onClick={[Function]}
      />
      <div
        className="movie-list__no-response_active"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
